airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/errorreporter/JobErrorReporter.java

```diff
@@ -82,7 +83,9 @@ public JobErrorReporter(final ConfigRepository configRepository,
   public void reportSyncJobFailure(final UUID connectionId, final AttemptFailureSummary failureSummary, final SyncJobReportingContext jobContext) {
     Exceptions.swallow(() -> {
       final List<FailureReason> traceMessageFailures = failureSummary.getFailures().stream()
-          .filter(failure -> failure.getMetadata() != null && failure.getMetadata().getAdditionalProperties().containsKey(FROM_TRACE_MESSAGE))
+          .filter(failure -> failure.getMetadata() != null &&
+              failure.getMetadata().getAdditionalProperties().containsKey(FROM_TRACE_MESSAGE) &&
+              failure.getFailureType() != FailureType.CONFIG_ERROR)
```

Comment:
Do we also want this to be the case for check/spec/discover failures? If so, I think a better spot might be in the `reportJobFailureReason` method, which they all end up calling.

Also, I would expect tests to need an update after this change.

Reply:
Agree on the tests!


I /think/ that this method isn't called by CHECK/SPEC/DISCOVER though.
* `jobErrorReporter.reportSyncJobFailure` seems only be called in [jobCreationAndStatusUpdateActivity](https://github.com/airbytehq/airbyte/blob/master/airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/JobCreationAndStatusUpdateActivityImpl.java#L270-L271) which in turn calls this method ([here](https://github.com/airbytehq/airbyte/blob/master/airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/JobCreationAndStatusUpdateActivityImpl.java#L270-L271))
* These other methods are for [checks](https://github.com/airbytehq/airbyte/blob/370feccb51bf0a5cbba311c27c31c3a837300469/airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/errorreporter/JobErrorReporter.java#L156-L167) and [discover](https://github.com/airbytehq/airbyte/blob/370feccb51bf0a5cbba311c27c31c3a837300469/airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/errorreporter/JobErrorReporter.java#L176-L187) errors

Reply:
I thought Pedro's point was should this filtering also occur for `check`, `spec`, `discover` and if so then the filtering should also occur in `reportJobFailureReason` so it gets filtered for all those phases

I guess I'm also curious on the difference between `reportJobFailureReason` and `reportSyncJobFailure`, is `Job` a superset of `Sync` and so `Job` is better since it will encompass this filtering better or something else altogether?

Reply:
> I think that this method isn't called by CHECK/SPEC/DISCOVER though.

@evantahler  Yeah, that's what I was trying to say, that it should be done in `reportJobFailureReason` rather than `reportSyncJobFailure`. Akash's PR is using the former, so we should be good on this front.

> I guess I'm also curious on the difference between reportJobFailureReason and reportSyncJobFailure, is Job a superset of Sync and so Job is better since it will encompass this filtering better or something else altogether?

@ryankfu Yeah, in this context "job" refers to any of the jobs we run on connectors, e.g. check, spec, discover, sync, etc. There are specific methods to report a failure reason for each job type because there are nuances to what data we have and how we build up the metadata, but they all ultimately call the general reportJobFailureReason which does some more common things and actually calls on the sentry client